rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -1339,6 +1327,27 @@ public InputStream jobFullLogs(@HeaderParam("sessionid") String sessionId, @Path
         }
     }
 
+    protected InputStream retrieveTaskLogs(TaskState taskState, String sessionId, String jobId) {
+        InputStream inputStream = null;
+        try {
+            if (taskState.isPreciousLogs()) {
+                inputStream = retrieveTaskLogsUsingDataspaces(sessionId, jobId, taskState.getId());
+            } else {
+                String taskLogs = retrieveTaskLogsUsingDatabase(sessionId, jobId, taskState.getName());
+
+                if (!taskLogs.isEmpty()) {
+                    inputStream = IOUtils.toInputStream(taskLogs, Charset.forName(FILE_ENCODING));
+                }
+
+                logger.warn("Retrieving truncated logs for task '" + taskState.getId() + "'");
+            }
+        } catch (Exception e) {
+            logger.info("Could not retrieve logs for task " + taskState.getId() +
+                        " (could be a non finished or killed task)", e);
+        }
+        return inputStream;
+    }
+
     public InputStream retrieveTaskLogsUsingDataspaces(String sessionId, String jobId, TaskId taskId)
             throws PermissionRestException, IOException, NotConnectedRestException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)